cryptography/src/bls12381/benches/dkg_recovery.rs

```diff
@@ -26,35 +24,39 @@ fn benchmark_dkg_recovery(c: &mut Criterion) {
             b.iter_batched(
                 || {
-                    // Create contributors
-                    let contributors = (0..n)
-                        .map(|i| ed25519::PrivateKey::from_seed(i as u64).public_key())
+                    let mut private_keys = (0..n)
+                        .map(|i| PrivateKey::from_seed(i as u64))
+                        .collect::<Vec<_>>();
+                    let me = private_keys.pop().unwrap();
+                    let me_pk = me.public_key();
+                    let dealers = private_keys
+                        .iter()
+                        .map(|sk| sk.public_key())
                         .collect::<Ordered<_>>();
+                    let players = Ordered::from_iter(iter::once(me_pk.clone()));
+                    let round_info = RoundInfo::new(0, None, dealers, players).unwrap();
 
                     // Create player
-                    let me = contributors[0].clone();
-                    let mut player = Player::<_, MinSig>::new(
-                        me,
-                        None,
-                        contributors.clone(),
-                        contributors.clone(),
-                        CONCURRENCY,
-                    );
+                    let mut player =
+                        Player::<MinSig, PrivateKey>::new(round_info.clone(), me).unwrap();
 
                     // Create commitments and send shares to player
-                    let mut commitments = BTreeMap::new();
-                    for (idx, dealer) in contributors.iter().take(t as usize).enumerate() {
-                        let (_, commitment, shares) =
-                            Dealer::<_, MinSig>::new(&mut rng, None, contributors.clone());
-                        player
-                            .share(dealer.clone(), commitment.clone(), shares[0].clone())
+                    let mut logs = BTreeMap::new();
+                    for sk in private_keys {
+                        let pk = sk.public_key();
+                        let (mut dealer, pub_msg, mut priv_msgs) =
+                            Dealer::start(&mut rng, round_info.clone(), sk, None).unwrap();
+                        let ack = player
+                            .dealer_message(pk, pub_msg, priv_msgs.pop().unwrap().1)
                             .unwrap();
-                        commitments.insert(idx as u32, commitment);
+                        dealer.receive_player_ack(me_pk.clone(), ack).unwrap();
+                        let (checked_pk, log) = dealer.finalize().check(&round_info).unwrap();
+                        logs.insert(checked_pk, log);
                     }
-                    (player, commitments)
+                    (player, logs)
                 },
-                |(player, commitments)| {
-                    black_box(player.finalize(commitments, BTreeMap::new()).unwrap());
+                |(player, logs)| {
+                    black_box(player.finalize(logs).unwrap());
```

Comment:
Do we offer concurrency on `player` still? I found share recovery to take the longest of all ops?

Reply:
No, we should add this back again.